class-vip-support-user.php

```diff
@@ -400,6 +418,62 @@ public function filter_wp_redirect( $location ) {
 		return $location;
 	}
 
+	/**
+	 * Filters a users email address, removing the username if the email
+	 * is a VIP Support email alias.
+	 *
+	 * This helps simplify support email aliases in the users list so
+	 * they're not as lengthy and easier to understand at a glance.
+	 *
+	 * @param $email string The email address of a user.
+	 *
+	 * @return string
+	 */
+	public function filter_vip_support_email_aliases( $email ) {
+		if ( is_admin() && $this->is_vip_support_email_alias( $email ) ) {
+			return self::VIP_SUPPORT_EMAIL_ADDRESS;
+		}
+		return $email;
+	}
+
+	/**
+	 * Filters a users Gravatar URL, altering it if the email is a VIP
+	 * Support email alias.
+	 *
+	 * Since the email aliases are fake, they don't have a Gravatar.
+	 * Instead of showing the mystery man, get the Gravatar for the
+	 * real `VIP_SUPPORT_EMAIL_ADDRESS` email address.
+	 *
+	 * @param $url         string The Gravatar url.
+	 * @param $id_or_email mixed  The users ID, email, or a WP_User, WP_Post, WP_Comment object.
+	 * @param $args        array  Arguments passed to get_avatar_url(), after processing.
+	 *
+	 * @return string
+	 */
+	public function filter_vip_support_email_gravatars( $url, $id_or_email, $args ) {
+
+		if ( ! is_admin() ) {
+			return $url;
+		}
+
+		// Get the user's email address.
+		if ( is_numeric( $id_or_email ) ) {
```

Comment:
This conditional block could be abstracted into it's own private method to make what's happening here more obvious. 

```php
$user_email = $this->get_user_email_from_id_or_email( $id_or_email );
```

```php
private function get_user_email_from_id_or_email( $id_or_email ) {
	if ( is_numeric( $id_or_email ) ) {
		$user = get_user_by( 'id', $id_or_email );
		$user_email = $user->user_email;
	} elseif ( is_string( $id_or_email ) ) {
		$user_email = $id_or_email;
	} elseif ( $id_or_email instanceof WP_User ) {
		$user_email = $id_or_email->user_email;
	}

	return '';
}
```

While the method could return `null`, I'd rather see it keep the returned string types consistent - the `isset( $user_email )` check below could be changed to `! empty( $user_email )`.